node/src/components/consensus/era_supervisor.rs

```diff
@@ -1009,6 +1009,10 @@ where
                 equivocators,
                 proposer,
             }) => {
+                if era_id != self.era_supervisor.current_era {
+                    debug!(era = era_id.value(), "finalized block in old era");
```

Comment:
Shouldn't this be a warn or err? After all, this should never happen and it would be a bug.

Reply:
No, it can definitely happen: If we just joined, and initialized eras 1, 2 and 3, we'll simultaneously download their protocol states, and see blocks finalized in all of them.

It's just that we can safely ignore that, because if we already initialized era 3 that means we have all of era 2's blocks in the linear chain.

Reply:
Ah, of course. Thanks.